File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        std::set<int> set;
        vector<int> res;
        
        for(int num: nums){
            if(set.contains(num)){
                res.push_back(num);
            } else{
                set.insert(num);                
            }
        }
        
        
        return res;
    }
};

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        Set<Integer> set = new HashSet<>();
        List<Integer> lst = new ArrayList<>();
        
         for(int num: nums){
            if(set.contains(num)){
                lst.add(num);
            } else{
                set.add(num);                
            }
        }
        
        return lst;
    }
}